api/middleware/contentLengthLimiterMiddleware.go

```diff
@@ -0,0 +1,59 @@
+package middleware
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/multiversx/mx-chain-go/api/shared"
+	"net/http"
+)
+
+type contentLengthLimiterMiddleware struct {
+	MaxSizeBytes int64
+}
+
+// NewContentLengthLimiterMiddleware will abort all requests that have Content-Length size bigger
+// than the one specified in config.
+func NewContentLengthLimiterMiddleware(maxSizeBytes int64) *contentLengthLimiterMiddleware {
+	return &contentLengthLimiterMiddleware{MaxSizeBytes: maxSizeBytes}
+}
+
+// MiddlewareHandlerFunc returns the handler func used by the gin server when processing requests.
+func (r *contentLengthLimiterMiddleware) MiddlewareHandlerFunc() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		size := c.Request.ContentLength
+
+		if size == -1 {
+			log.Debug("unknown content length", "error", errors.New("content length is -1"))
```

Comment:
I added the uknown use case, because of this.

// ContentLength records the length of the associated content. // The value -1 indicates that the length is unknown. // Values >= 0 indicate that the given number of bytes may // be read from Body. // // For client requests, a value of 0 with a non-nil Body is // also treated as unknown. ContentLength [int64]

https://pkg.go.dev/net/http#Request.ContentLength